src/main/java/com/google/sps/utils/data/AnimalHistory.java

```diff
@@ -0,0 +1,71 @@
+package com.google.sps.data;
+
+public class AnimalHistory {
+    private int year;
+    private String code;
+    private String category;
+
+    private AnimalHistory() {
+        year = 0;
+        code = null;
+        category = null;
+    }
+
+    private AnimalHistory(int year, String code, String category) {
+        this.year = year;
+        this.code = code;
+        this.category = category;
+    }
+
+    public void setYear(int year) {
+        this.year = year;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    public void setCategory(String code) {
+        this.category = category;
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = year;
+        final int prime = 31;
+
+        hash = hash * prime + code.hashCode();
+        hash = hash * prime + category.hashCode();
+        return hash;
```

Comment:
I think one of these `hash = ...` lines should be commented/removed? Or maybe combine the hash into `hash = hash * prime + code.hashCode() + category.hashCode();`?